gradle/npm.gradle

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+// The webpack tasks are configured to output build artifacts to $projectDir/npmdist
+// (configured in build.gradle via js.browser.distribution.directory)
+// This task uses $projectDir/npm/package.json as a template to build the final 
+// $projectDir/npmdist/package.json which can then be published.
+//
+// Test with:
+// ./gradlew clean
+// ./gradlew jsBrowserDevelopmentWebpack
+// ./gradlew publishNpm -P dryRun=true -P npmPublishVersion=0.0.3-test5
+// Publish:
+// ./gradlew clean
+// ./gradlew jsBrowserProductionWebpack
+// ./gradlew publishNpm -P kotlin.npmjs.auth.token=<token>
+
+
+def prop(name, defVal) {
+    def value = project.properties[name]
+    if (value == null) return defVal
+    return value
+}
+
+def distTag(version) {
+    def i = version.indexOf('-')
+    if (i > 0) return version.substring(i + 1)
+    return "latest"
+}
+
+def getLatestGitTag() {
+    def proc = "gradle/gitLatestTag.sh".execute()
+    def outputStream = new StringBuffer()
+    proc.waitForProcessOutput(outputStream, System.err)
+    return outputStream.toString().replaceAll("\\s","")
+}
+
+def authToken = prop("kotlin.npmjs.auth.token", "")
+def dryRun = prop("dryRun", "true")
+
+
+ext {
+    // These are referenced from the package.json template
+    kotlinVersion = "1.4.10"
+    npmPublishVersion = prop("npmPublishVersion", getLatestGitTag())
+}
```

Comment:
These are not actually used. Anything defined outside of a task is no longer available when using the Kotlin DSL. I don't know how to resolve this problem.